wgpu-types/src/lib.rs

```diff
@@ -711,89 +928,160 @@ impl CompareFunction {
     Ushort4Norm = 13,
+    /// Two signed shorts (i16). [-32767, 32767] converted to float [-1, 1] `vec2` in shaders.
     Short2Norm = 14,
+    /// Four signed shorts (i16). [-32767, 32767] converted to float [-1, 1] `vec4` in shaders.
     Short4Norm = 15,
+    /// Two half-precision floats (no Rust equiv). `vec2` in shaders.
     Half2 = 16,
+    /// Four half-precision floats (no Rust equiv). `vec4` in shaders.
     Half4 = 17,
+    /// One single-precision float (f32). `float` in shaders.
     Float = 18,
+    /// Two single-precision floats (f32). `vec2` in shaders.
     Float2 = 19,
+    /// Three single-precision floats (f32). `vec3` in shaders.
     Float3 = 20,
+    /// Four single-precision floats (f32). `vec4` in shaders.
     Float4 = 21,
+    /// One unsigned int (u32). `uint` in shaders.
     Uint = 22,
+    /// Two unsigned ints (u32). `uvec2` in shaders.
     Uint2 = 23,
+    /// Three unsigned ints (u32). `uvec3` in shaders.
     Uint3 = 24,
+    /// Four unsigned ints (u32). `uvec4` in shaders.
     Uint4 = 25,
+    /// One signed int (i32). `int` in shaders.
     Int = 26,
+    /// Two signed ints (i32). `ivec2` in shaders.
     Int2 = 27,
+    /// Three signed ints (i32). `ivec3` in shaders.
     Int3 = 28,
+    /// Four signed ints (i32). `ivec4` in shaders.
     Int4 = 29,
 }
 
 bitflags::bitflags! {
+    /// Different ways that you can use a buffer.
+    ///
+    /// The usages determine what kind of memory the buffer is allocated from and what
+    /// actions the buffer can partake in.
     #[repr(transparent)]
     #[cfg_attr(feature = "trace", derive(Serialize))]
     #[cfg_attr(feature = "replay", derive(Deserialize))]
     pub struct BufferUsage: u32 {
+        /// Allow a buffer to be mapped for reading using `map_read`. This does not include
+        /// creating a buffer with `mapped_at_creation` set.
+        ///
+        /// If [`Extensions::MAPPABLE_PRIMARY_BUFFERS`] isn't enabled, the only other usage a buffer
+        /// may have is COPY_DST.
         const MAP_READ = 1;
+        /// Allow a buffer to be mapped for writing using `map_write`. This does not include
+        /// creating a buffer with `mapped_at_creation` set.
+        ///
+        /// If [`Extensions::MAPPABLE_PRIMARY_BUFFERS`] extension isn't enabled, the only other usage a buffer
+        /// may have is COPY_SRC.
         const MAP_WRITE = 2;
+        /// Allow a buffer to be the source buffer for a `copy_buffer_to_buffer` or `copy_buffer_to_texture`
+        /// operation.
         const COPY_SRC = 4;
+        /// Allow a buffer to be the destination buffer for a `copy_buffer_to_buffer` or `copy_texture_to_buffer`
+        /// operation.
         const COPY_DST = 8;
+        /// Allow a buffer to be the index buffer in a draw operation.
         const INDEX = 16;
+        /// Allow a buffer to be the vertex buffer in a draw operation.
         const VERTEX = 32;
+        /// Allow a buffer to be a uniform buffer inside a bind group.
```

Comment:
could list the actual `BindingType` variants supported for these, as a follow-up